manifests/pam.pp

```diff
@@ -131,6 +133,32 @@
         }
       }
 
+      #only allow root and members of the group wheel to su
+      if $only_root_may_su {
+        case $::operatingsystem {
```

Comment:
this `case` isn't working here, because you'll never reach this if you are not debian or ubuntu --> see line 48
So the whole block must be copied, customized and pasted into the end / line 167

I see this is hard to setup & test if you don't have access to those environments ... let me suggest this: reduce your solution once again (sorry) to ubuntu and I'll adapt this for the other platforms and insert another pr myself afterwards.

Reply:
Yes, sorry, I'm only testing on Debian Stretch due to time constraints :/ I'm testing my work on a modified version of https://github.com/dev-sec/hardening/tree/master/puppet (as I want to use the contrib-stretch image, for instance). If there's a similar solution for RedHat/Suse, I'm happy to add it to my testing before I send in my patches! It's just that RedHat and Suse are fairly unknown to me (been using Debian since 1997), so it's not that easy for me to setup (and time does not permit it either).

For now, I'll do as requested and move it down.

Reply:
s/move it down/reduce to debian/